src/core/thread/osthread.d

```diff
@@ -3182,23 +3182,42 @@ private void onThreadError(string msg) nothrow @nogc
 }
 
 version (Posix)
-private size_t adjustStackSize(size_t sz) nothrow @nogc
 {
-    if (sz == 0)
-        return 0;
-
-    version (CRuntime_Glibc)
+    private size_t adjustStackSize(size_t sz) nothrow @nogc
     {
-        // TLS uses the top of the stack, so add its size to the requested size
-        sz += externDFunc!("rt.sections_elf_shared.sizeOfTLS",
-                           size_t function() @nogc nothrow)();
+        if (sz == 0)
+            return 0;
+
+        version (CRuntime_Glibc)
+        {
+            // TLS uses the top of the stack, so add its size to the requested size
+            sz += externDFunc!("rt.sections_elf_shared.sizeOfTLS",
+                               size_t function() @nogc nothrow)();
+        }
+        // stack size must be a multiple of PAGESIZE and at least PTHREAD_STACK_MIN
+        sz = ((sz + PAGESIZE - 1) & ~(PAGESIZE - 1));
+
+        size_t minStackSize = minimalStackSize();
```

Comment:
Instead of evaluating sizeOfTLS size twice, it should be simpler and more efficient to save it to a variable and adjust `sz` and `minStackSize` accordingly inside this function.

Reply:
Thank you for notice that. How stupid I was...